addons/medical_engine/functions/fnc_getItemArmour.sqf

```diff
@@ -44,10 +46,10 @@ if (!isClass _protection) exitwith {
 private _armourValues = [];
 {
     private _protectionHitpoint = toLower getText (_x >> "hitpointName");
-    private _armour = if (isNumber (_x >> "armor")) then {getNumber (_x >> "armor")} else {0};
-    private _expl = if (isNumber (_x >> "explosionShielding")) then {getNumber (_x >> "explosionShielding")} else {0};
+    private _armour = getNumber (_x >> "armor");
+    private _expl = getNumber (_x >> "explosionShielding");
     _armourValues pushBack [_protectionHitpoint,[_armour,_expl]];
-} foreach ("isText (_x >> ""hitpointName"")" configClasses _protection);
+} forEach configProperties [_protection, "isClass _x && {isText (_x >> ""hitpointName"")}"];
```

Comment:
Wait, why would there be a sub-class without hitpointName entry?

Reply:
Tested and it seems that if you do have one the vest is broken, the game won't even create the item; so I guess it's not an issue.